services/terraformer/server/clusterBuilder/clusterBuilder.go

```diff
@@ -158,34 +158,35 @@ func (c ClusterBuilder) DestroyNodepools() error {
 	return nil
 }
 
-func (c ClusterBuilder) generateFiles(clusterID, clusterDir string) error {
-	// generate backend
-	backend := backend.Backend{ProjectName: c.ProjectName, ClusterName: clusterID, Directory: clusterDir}
-	err := backend.CreateFiles()
-	if err != nil {
+func (c *ClusterBuilder) generateFiles(clusterID, clusterDir string) error {
+	backend := backend.Backend{
+		ProjectName: c.ProjectName,
+		ClusterName: clusterID,
+		Directory:   clusterDir,
+	}
+
+	if err := backend.CreateFiles(); err != nil {
 		return err
 	}
 
-	// generate .tf files for nodepools
-	var clusterInfo *pb.ClusterInfo
-	template := templateUtils.Templates{Directory: clusterDir}
-	templateLoader := templateUtils.TemplateLoader{Directory: templateUtils.TerraformerTemplates}
+	// generate Providers terraform configuration
+	providers := provider.Provider{
+		ProjectName: c.ProjectName,
+		ClusterName: clusterID,
+		Directory:   clusterDir,
+	}
 
+	if err := providers.CreateProvider(c.CurrentInfo, c.DesiredInfo); err != nil {
```

Comment:
Do we need to check whether `CurrentInfo` or `DesiredInfo` is not `nil` before calling this function?

Reply:
No, it will be checked in the `getProvidersUsed` func

https://github.com/berops/claudie/blob/93402dd20ecfec0335716b3bb7e865f8e013a004/services/terraformer/server/provider/provider.go#L62-L65